File: src/easy/leetcode_706.py
class Bucket:
    def __init__(self):
        self.data = []

    def put(self, key: int, value: int) -> None:
        index = -1

        for i, e in enumerate(self.data):
            if e[0] == key:
                index = i
                break

        if index == -1:
            self.data.append((key, value))
        else:
            self.data[index] = (key, value)

    def get(self, key: int) -> int:
        for i, e in enumerate(self.data):
            if e[0] == key:
                return e[1]
        return -1

    def remove(self, key: int) -> None:
        for i, e in enumerate(self.data):
            if e[0] == key:
                self.data.remove(e)
                return





class MyHashMap:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.data = []
        self.key_size = 2069
        for i in range(self.key_size):
            self.data.append(Bucket())

    def put(self, key: int, value: int) -> None:
        """
        value will always be non-negative.
        """
        hash = key % self.key_size
        bucket = self.data[hash]
        bucket.put(key, value)

    def get(self, key: int) -> int:
        """
        Returns the value to which the specified key is mapped, or -1 if this map contains no mapping for the key
        """
        hash = key % self.key_size
        bucket = self.data[hash]
        return bucket.get(key)


    def remove(self, key: int) -> None:
        """
        Removes the mapping of the specified value key if this map contains a mapping for the key
        """
        hash = key % self.key_size
        bucket = self.data[hash]
        return bucket.remove(key)

File: src/easy/leetcode_706.java
package easy;

import java.util.ArrayList;
import java.util.LinkedList;

public class leetcode_706 {


    class Pair<U, V> {
        U key;
        V value;
        Pair(U key, V value) {
            this.key = key;
            this.value = value;
        }
    }


    class Bucket {
        ArrayList<Pair<Integer, Integer>> data;

        Bucket() {
            this.data = new ArrayList<Pair<Integer, Integer>>();
        }

        public void update(Pair<Integer, Integer> pair) {
            boolean found = false;
            for(Pair<Integer, Integer> p: this.data) {
                if(p.key == pair.key) {
                    found = true;
                    p.value = pair.value;
                    return;
                }
            }
            if(!found) {
                this.data.add(pair);
            }
        }

        public int get(int key) {
            for(Pair<Integer, Integer> p: this.data) {
                if(p.key == key) {
                    return p.value;
                }
            }
            return -1;
        }

        public void remove(int key) {
            for(Pair<Integer, Integer> p: this.data) {
                if(p.key == key) {
                    this.data.remove(p);
                    return;
                }
            }
        }



    }


    class MyHashMap {

        Bucket[] data;
        int key_size;

        /** Initialize your data structure here. */
        public MyHashMap() {
            this.key_size = 2069;
            this.data = new Bucket[this.key_size];
            for(int i=0;i<this.key_size;i++) {
                this.data[i] = new Bucket();
            }
        }

        /** value will always be non-negative. */
        public void put(int key, int value) {
            int hash = key % this.key_size;
            Bucket b = this.data[hash];
            b.update(new Pair<Integer, Integer>(key, value));
        }

        /** Returns the value to which the specified key is mapped, or -1 if this map contains no mapping for the key */
        public int get(int key) {
            int hash = key % this.key_size;
            Bucket b = this.data[hash];
            return b.get(key);
        }

        /** Removes the mapping of the specified value key if this map contains a mapping for the key */
        public void remove(int key) {
            int hash = key % this.key_size;
            Bucket b = this.data[hash];
            b.remove(key);
        }
    }


    public void solve() {

    }

    public static void main(String[] args) {
        leetcode_706 t = new leetcode_706();
        t.solve();
    }
}
